intro-to-R-tidyverse/02-intro_to_ggplot2.Rmd

```diff
@@ -0,0 +1,258 @@
+head(stats_df$contrast)
+```
+
+If we want to see less values at once, we can use brackets with the indices of the values we'd like returned.
+
+```{r}
+stats_df$avg_expression[1:10]
+```
+
+Let's checkout how many rows of data there are per each contrast.
+
+```{r}
+summary(as.factor(stats_df$contrast))
+```
+
+## Set up the dataset
+
+Make a new variable of -log10 p values and for that we'll use `dplyr::mutate`.
+
+```{r}
+stats_df <- mutate(stats_df, # data.frame we'd like to add a variable to
+                   neg_log_p_val = -log10(p_value))
+```
+
+Let's filter to only `male_female` contrast data. 
+First let's try out a logical expression: 
+
+```{r}
+stats_df$contrast == "male_female"
+```
+
+Notice that we are not reassigning here, so this filtered dataset will not be saved to the environment
+
+```{r}
+filter(stats_df, contrast == "male_female")
+```
+
+Now we can reassign to `male_female_df`. 
+
+```{r}
+male_female_df <- filter(stats_df, contrast == "male_female")
+```
+
+## Plotting this data
+
+Let's make a volcano plot with this data. 
+First let's take a look at only the tumor vs normal data. 
+Let's save this as a separate data.frame by assigning it a new name. 
+
+```{r}
+tumor_normal_df <- filter(stats_df, contrast == "astrocytoma_normal") 
+```
+
+We will use this data.frame in `ggplot` function. 
+
+```{r}
+ggplot(tumor_normal_df, # This first part has the data.frame of data we want to plot
+       aes(x = log_fold_change, # This is the column name of the data we want to be the x-axis
+           y = neg_log_p_val) # This is the column name of the data we want for the y-axis
+       )
+```
+
+You'll notice this plot doesn't have anything on it because we haven't 
+specified a plot type yet. 
+To do that, we will add another ggplot layer with `+`. 
+
+```{r}
+# This first part is the same as before
+ggplot(tumor_normal_df,
+       aes(x = log_fold_change, 
+           y = neg_log_p_val)) + 
+  # Now we are adding on a layer to specify what kind of plot we want
+  geom_point()
+```
+
+Here's a brief summary of ggplot2 structure. 
+[ggplot2 structure](diagrams/ggplot_structure.png)
+
+### Adjust our ggplot
+
+Now that we have a base plot that shows our data, we can add layers on to it and adjust it.
+We can adjust the color points using `color`. 
+
+```{r}
+ggplot(tumor_normal_df, 
+       aes(x = log_fold_change,
+           y = neg_log_p_val,
+           color = avg_expression) # We added this argument to color code the points!
+       ) + 
+  geom_point()
+```
+
+Because we have so many points overlapping one another, we will want to adjust 
+the transparency, which we can do with an `alpha` argument. 
+
+```{r}
+ggplot(tumor_normal_df, 
+       aes(x = log_fold_change,
+           y = neg_log_p_val,
+           color = avg_expression)
+       ) + 
+  geom_point(alpha = 0.2) # We are using the `alpha` argument to make our points transparent
+```
+
+We can also change the background and aesthetics by adding a `theme`.
```

Comment:
Trying to avoid "aesthetic" except in the context of `aes()`
```suggestion
We can also change the background and  appearance of the plot as a whole by adding a `theme`.
```